build_tools/travis/install.sh

```diff
@@ -95,18 +95,26 @@ if [[ "$COVERAGE" == "true" ]]; then
     pip install coverage coveralls
 fi
 
-if [ ! -d "$CACHED_BUILD_DIR" ]; then
-    mkdir -p $CACHED_BUILD_DIR
-fi
+if [[ "$SKIP_TESTS" == "true" ]]; then
+    echo "No need to build scikit-learn when not running the tests"
+else
+    if [ ! -d "$CACHED_BUILD_DIR" ]; then
+        mkdir -p $CACHED_BUILD_DIR
+    fi
 
-rsync -av --exclude '.git/' --exclude='testvenv/' \
-      $TRAVIS_BUILD_DIR $CACHED_BUILD_DIR
+    rsync -av --exclude '.git/' --exclude='testvenv/' \
+          $TRAVIS_BUILD_DIR $CACHED_BUILD_DIR
 
-cd $CACHED_BUILD_DIR/scikit-learn
+    cd $CACHED_BUILD_DIR/scikit-learn
 
-# Build scikit-learn in the install.sh script to collapse the verbose
-# build output in the travis output when it succeeds.
-python --version
-python -c "import numpy; print('numpy %s' % numpy.__version__)"
-python -c "import scipy; print('scipy %s' % scipy.__version__)"
-python setup.py develop
+    # Build scikit-learn in the install.sh script to collapse the verbose
+    # build output in the travis output when it succeeds.
+    python --version
+    python -c "import numpy; print('numpy %s' % numpy.__version__)"
+    python -c "import scipy; print('scipy %s' % scipy.__version__)"
+    python setup.py develop
+fi
+
+if [[ "$RUN_FLAKE8" == "true" ]]; then
```

Comment:
I know nothing about bash booleans in case they do exist. Is there any clear advantage of using them? Also I guess INSTALL_MKL was done used like this so I just copied it.